### PR TITLE
feat: add eslint

### PR DESCRIPTION
**Pull request by ZEROM22**

### CODE REVIEW

File: makefile.js [hunk 1]
```diff
@@ -282,4 +282,19 @@ app
...
+      const { fix } = options
+      let eslintCommand = "npx eslint ."
+      if (fix) eslintCommand = "npx eslint . --fix"
+      exec(eslintCommand)
+    } catch {
+      error("ESLint failed. Please fix the linting errors.")
+      exit(1)
+    }
+  })
+
```
Comment (vanyauhalin) [inline makefile.js:299]: You don't have to wrap the ESLint commands in the Makefile. Simply add `eslint` to the `package.json` scripts:

```sh
$ cat ./package.json | jq .scripts
{
  "lint": "eslint ."
}
```

Then execute it:

```sh
$ pnpm lint
# or
$ pnpm lint --fix
```

File: .eslintrc.json
Comment (vanyauhalin) [inline .eslintrc.json:1]: Their comments don't reflect your personal opinion or provide any additional information. They simply repeat everything that can be found in the ESLint's documentation. I recommend deleting them.

**Review by vanyauhalin — CHANGES_REQUESTED** [review]

**Comment by vanyauhalin** [comment]
Additionally, please rename the `.eslintrc.json` to `eslint.config.js`. Also, add the `@stylistic/eslint-plugin-js` as a dependency.

Read more about the file [here](https://eslint.org/docs/latest/use/configure/configuration-files-new), and about the dependency [here](https://eslint.style/guide/why).

**Comment by vanyauhalin** [comment]
> Additionally, please rename the `.eslintrc.json` to `eslint.config.js`. Also, add the `@stylistic/eslint-plugin-js` as a dependency.
> 
> Read more about the file [here](https://eslint.org/docs/latest/use/configure/configuration-files-new), and about the dependency [here](https://eslint.style/guide/why).

And please migrate to the flat config. Read more [here](https://eslint.org/docs/latest/extend/plugin-migration-flat-config).

**Comment by ZEROM22** [comment]
> Additionally, please rename the `.eslintrc.json` to `eslint.config.js`. Also, add the `@stylistic/eslint-plugin-js` as a dependency.
> 
> Read more about the file [here](https://eslint.org/docs/latest/use/configure/configuration-files-new), and about the dependency [here](https://eslint.style/guide/why).

I added `@stylistic/eslint-plugin-js' and rewrote the config to flat.

**Review by vanyauhalin — APPROVED** [review]

**Comment by vanyauhalin** [comment]
@ZEROM22, please resolve conflicts.

**Pull request merged**